litevideo/csc/rgb2rgb16f.py

```diff
@@ -0,0 +1,93 @@
+# rgb2rgb16f
+
+from litex.gen import *
+from litex.soc.interconnect.stream import *
+
+from litevideo.csc.common import *
+
+@CEInserter()
+class PIX2PIXFDatapath(Module):
+    """ 
+    Converts a 8 bit unsigned int represented by a pixel in 
+    the range [0-255] to a 16 bit half precision floating point 
+    pix_number defined in the range [0-1] 
+    """
+    latency = 2
+
+    def __init__(self, pix_w, pixf_w):
+        self.sink = sink = Record(pix_layout(pix_w))
+        self.source = source = Record(pixf_layout(pixf_w))
+
+        # # #
+
+        # delay pix signal
+        pix_delayed = [sink]
+        for i in range(self.latency):
+            pix_n = Record(pix_layout(pix_w))
+            self.sync += getattr(pix_n, "pix").eq(getattr(pix_delayed[-1], "pix"))
+            pix_delayed.append(pix_n)
+
+        # Hardware implementation:
+
+        # Stage 1
+        # Leading one detector
+
+        lshift = Signal(4)
+        frac_val = Signal(10)
+
+        self.sync += [
+            # Leading one detector
+            If( sink.pix[7]==1,
+                lshift.eq(0)
+            ).Elif(sink.pix[6] == 1,
+                lshift.eq(1)
+            ).Elif(sink.pix[5] == 1,
+                lshift.eq(2)
+            ).Elif(sink.pix[4] == 1,
+                lshift.eq(3)
+            ).Elif(sink.pix[3] == 1,
+                lshift.eq(4)
+            ).Elif(sink.pix[2] == 1,
+                lshift.eq(5)
+            ).Elif(sink.pix[1] == 1,
+                lshift.eq(6)
+            ).Elif(sink.pix[0] == 1,
+                lshift.eq(7)
+            ).Else(
+                lshift.eq(14)   #Zero
+            ),
+
+            frac_val[3:].eq(sink.pix[:7]),
+            frac_val[:3].eq(0)
+        ]
+
+        # Stage 2
+        # Adjust frac and exp components as per lshift
+        # Pack in 16bit float
+
+        self.sync += [
+            source.pixf[:10].eq(frac_val << lshift),
```

Comment:
Not tested on hardware. I viewed the technology schematic of synthesis results from Verilog file generated by Migen shifter code. I still need to add some overheads (adding CSR and relevant functions in ci.c) to test this on hardware.